arcgis-java-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/UtilityNetworkTraceTool.java

```diff
@@ -0,0 +1,999 @@
+  }
+
+  /**
+   * Returns a read-only property containing the last completed trace as a {@link UtilityNetworkTraceToolCompletedTrace}
+   * so that completed traces can be observed and containing data accessed.
+   *
+   * @return a read-only property for the {@link UtilityNetworkTraceToolCompletedTrace}
+   * @since 100.15.0
+   */
+  public ReadOnlyObjectProperty<UtilityNetworkTraceToolCompletedTrace> completedTraceReadOnlyProperty() {
+    return completedTraceProperty.getReadOnlyProperty();
+  }
+
+  /**
+   * Property that determines whether a MapView's onMouseClicked event is used for adding starting points.
+   * Defaults to true.
+   *
+   * @return the isAddingStartingPoints property
+   * @since 100.15.0
+   */
+  public SimpleBooleanProperty isAddingStartingPointsProperty() { return isAddingStartingPointsProperty; }
+
+  /**
+   * Returns the value of the isAddingStartingPointsProperty which determines whether a MapView's onMouseClicked event
+   * is used for adding starting points.
+   *
+   * @return true if enabled, false otherwise
+   * @since 100.15.0
+   */
+  public Boolean getIsAddingStartingPoints() { return isAddingStartingPointsProperty.get(); }
+
+  /**
+   * Sets the value of the isAddingStartingPointsProperty which determines whether a MapView's onMouseClicked event
+   * is used for adding starting points.
+   *
+   * @param isAddingStartingPoints true to enable, false to disable
+   * @since 100.15.0
+   */
+  public void setIsAddingStartingPoints(Boolean isAddingStartingPoints) { isAddingStartingPointsProperty.set(isAddingStartingPoints); }
+
+  /**
+   * Property that determines whether the MapView's viewpoint should change to focus on trace results.
+   * Defaults to true.
+   *
+   * @return the autoZoomToResults property
+   * @since 100.15.0
+   */
+  public SimpleBooleanProperty autoZoomToResultsProperty() { return autoZoomToResultsProperty; }
+
+  /**
+   * Returns the value of the autoZoomToResultsProperty which determines whether the MapView's viewpoint should change
+   * to focus on trace results.
+   *
+   * @return true if enabled, false otherwise
+   * @since 100.15.0
+   */
+  public Boolean getAutoZoomToResults() { return autoZoomToResultsProperty.get(); }
+
+  /**
+   * Sets the value of the isAddingStartingPointsProperty which determines whether a MapView's onMouseClicked event
+   * is used for adding starting points.
+   *
+   * @param autoZoomToResults true to enable, false to disable
+   * @since 100.15.0
+   */
+  public void setAutoZoomToResults(Boolean autoZoomToResults) { autoZoomToResultsProperty.set(autoZoomToResults); }
+
+  /**
+   * Property that determines what symbol should be used to denote starting points for a trace.
+   * Default is a SimpleMarkerSymbol with a green cross.
+   *
+   * @return the startingPointSymbol property
+   * @since 100.15.0
+   */
+  public SimpleObjectProperty<Symbol> startingPointSymbolProperty() { return startingPointSymbolProperty; }
```

Comment:
Is `null` an ok value for this property? Same question applies to other symbol properties.

Reply:
Good point. Have blocked those.